app/utils/title-route-extensions.js

```diff
@@ -0,0 +1,81 @@
+import config from "../config/environment";
+import Ember from "ember";
+
+export var titleHistory = {};
+
+var replaceTitle;
+if (config.replaceTitle) {
+  replaceTitle = function replaceLocation(title){
+    window.document.title = title;
+  };
+} else {
+  replaceTitle = function replaceLocation(title){
+    titleHistory.last = title;
+  };
+}
+
+export { replaceTitle };
+
+// Via https://gist.github.com/machty/8413411
+
+// Extend Ember.Route to add support for sensible
+// document.title integration.
+Ember.Route.reopen({
+
+  // `titleToken` can either be a static string or a function
+  // that accepts a model object and returns a string (or array
+  // of strings if there are multiple tokens).
+  titleToken: null,
+
+  // `title` can either be a static string or a function
+  // that accepts an array of tokens and returns a string
+  // that will be the document title. The `collectTitleTokens` action
+  // stops bubbling once a route is encountered that has a `title`
+  // defined.
+  title: null,
+
+  // Provided by Ember
+  _actions: {
+    collectTitleTokens: function(tokens) {
+      var titleToken = this.titleToken;
+      if (typeof this.titleToken === 'function') {
+        titleToken = this.titleToken(this.currentModel);
+      }
+
+      if (Ember.isArray(titleToken)) {
+        tokens.unshift.apply(this, titleToken);
```

Comment:
and I may be reading this wrong but wouldn't it be `tokens.unshift.apply(tokens, titleToken)`?